Lib/test/datetimetester.py

```diff
@@ -61,7 +61,7 @@ def test_constants(self):
         self.assertEqual(datetime.MAXYEAR, 9999)
 
     def test_name_cleanup(self):
-        if '_Fast' not in str(self):
+        if self._test_type == 'Pure':
```

Comment:
In the past `__qualname__` didn't exist. `repr(self)` contained class' `__name__`. After introducing `__qualname__` that tests were not updated.

I think it would be better to modify both `__name__` and `__qualname__` (they should be consistent). `_test_type` is redundant since required information can be extracted from `__name__`.